### PR TITLE
feat(shell): count_data return estimate count by default

### PR DESCRIPTION
**Pull request by Shuo-Jia**
### What problem does this PR solve? 
 `count_data` need scan all online records to get precise result, which may affect
 cluster availability, so here define `precise = false` defaultly and it will return estimate
 count immediately.

### What is changed and how it works?
Define `precise = false` defaultly, if want to get precise result, user must add `[-c|--precise]` explictly.

### Check List 

Tests 
- Manual test (add detailed scripts or steps below)
```
// start onebox
>>> count_data
[count_data]
pidx       estimate_count  
0          14.00           
1          341.00          
2          381.00          
3          361.00          
4          30.00           
5          203.00          
6          63.00           
7          52.00           
(total:8)  1445.00    

>>> count_data -c
INFO: cluster_name = onebox
INFO: app_name = temp
INFO: partition = all
INFO: max_batch_count = 500
INFO: timeout_ms = 5000
INFO: hash_key_filter_type = no_filter
INFO: sort_key_filter_type = no_filter
INFO: value_filter_type = no_filter
INFO: diff_hash_key = false
INFO: stat_size = false
INFO: top_count = 0
INFO: run_seconds = 0
INFO: open app scanner succeed, partition_count = 8
INFO: prepare scanners succeed, split_count = 8
INFO: processed for 1 seconds, (8/8) splits, total 0 rows, last second 0 rows
INFO: split[0]: 0 rows
INFO: split[1]: 0 rows
INFO: split[2]: 0 rows
INFO: split[3]: 0 rows
INFO: split[4]: 0 rows
INFO: split[5]: 0 rows
INFO: split[6]: 0 rows
INFO: split[7]: 0 rows
Count done, total 0 rows.

>>> count_data --precise
INFO: cluster_name = onebox
INFO: app_name = temp
INFO: partition = all
INFO: max_batch_count = 500
INFO: timeout_ms = 5000
INFO: hash_key_filter_type = no_filter
INFO: sort_key_filter_type = no_filter
INFO: value_filter_type = no_filter
INFO: diff_hash_key = false
INFO: stat_size = false
INFO: top_count = 0
INFO: run_seconds = 0
INFO: open app scanner succeed, partition_count = 8
INFO: prepare scanners succeed, split_count = 8
INFO: processed for 1 seconds, (8/8) splits, total 0 rows, last second 0 rows
INFO: split[0]: 0 rows
INFO: split[1]: 0 rows
INFO: split[2]: 0 rows
INFO: split[3]: 0 rows
INFO: split[4]: 0 rows
INFO: split[5]: 0 rows
INFO: split[6]: 0 rows
INFO: split[7]: 0 rows
Count done, total 0 rows.

>>> count_data -b 10
ERROR: you must input [-c|--precise] flag when you expect to get pricise result by scaning all record online
````

Related changes

- Need to cherry-pick to the release branch **(Note: the pr depends #437)**
- Need to update the documentation
- Need to be included in the release note

### Note
If some key-value is duplicate(usually written multiple times), the value of key number will be count multi time, so only after compaction, the value just be relatively more accurate.

### CODE REVIEW

File: src/shell/commands/data_operations.cpp [hunk 1]
```diff
@@ -2201,6 +2211,44 @@ bool count_data(command_executor *e, shell_context *sc, arguments args)
...
+    // if input [-p|--partition] etc. means you want to get precise count by scanning, but no input
+    // [-c|--precise], it will return false
+    if (need_scan && !precise) {
+        fprintf(stderr,
+                "ERROR: you must input [-c|-pricise] flag when you expect to get pricise "
+                "result by scaning all record online\n");
+        return false;
+    }
+
+    if (!precise) {
```
Comment (levy5307) [inline src/shell/commands/data_operations.cpp:2223]: move these code here
```
    if (need_scan) {
        fprintf(stderr,
                "ERROR: you must input [-c|-pricise] flag when you expect to get pricise "
                "result by scaning all record online\n");
        return false;
    }
```

File: src/shell/commands/data_operations.cpp [hunk 2]
```diff
@@ -2201,6 +2211,44 @@ bool count_data(command_executor *e, shell_context *sc, arguments args)
...
+        if (!get_app_stat(sc, app_name, rows)) {
+            fprintf(stderr, "ERROR: query app stat from server failed");
+            return true;
+        }
+
+        rows.resize(rows.size() + 1);
+        row_data &sum = rows.back();
+        sum.row_name = "(total:" + std::to_string(rows.size() - 1) + ")";
+        for (int i = 0; i < rows.size() - 1; ++i) {
+            row_data &row = rows[i];
```
Comment (levy5307) [inline src/shell/commands/data_operations.cpp:2236]: `const row_data &row`

File: src/shell/commands/data_operations.cpp [hunk 3]
```diff
@@ -2201,6 +2211,44 @@ bool count_data(command_executor *e, shell_context *sc, arguments args)
...
+            sum.rdb_estimate_num_keys += row.rdb_estimate_num_keys;
+        }
+
+        ::dsn::utils::table_printer tp("count_data");
+        tp.add_title("pidx");
+        tp.add_column("estimate_count");
+        for (row_data &row : rows) {
```
Comment (levy5307) [inline src/shell/commands/data_operations.cpp:2243]: `const`

File: src/shell/commands/data_operations.cpp [hunk 4]
```diff
@@ -2201,6 +2211,44 @@ bool count_data(command_executor *e, shell_context *sc, arguments args)
         }
     }
 
+    if (!precise) {
+        // if input [-p|--partition] etc. means you want to get precise count by scanning, but no
```
Comment (acelyc111) [inline src/shell/commands/data_operations.cpp:2215]: Move this comment to line 2131 would be more clear.

File: src/shell/commands/data_operations.cpp [hunk 5]
```diff
@@ -2119,10 +2125,15 @@ bool count_data(command_executor *e, shell_context *sc, arguments args)
         int option_index = 0;
         int c;
         c = getopt_long(
-            args.argc, args.argv, "p:b:t:h:x:s:y:v:z:dan:r:", long_options, &option_index);
+            args.argc, args.argv, "cp:b:t:h:x:s:y:v:z:dan:r:", long_options, &option_index);
         if (c == -1)
             break;
+        // if input [-p|--partition] etc. means you want to get precise count by scanning.
```
Comment (neverchanje) [inline src/shell/commands/data_operations.cpp:2131]:
```suggestion
        // if specifying [-p|--partition] or [-c|--precise] means you want to get precise count by scanning.
```

File: src/shell/commands/data_operations.cpp [hunk 6]
```diff
@@ -2201,6 +2212,42 @@ bool count_data(command_executor *e, shell_context *sc, arguments args)
         }
     }
 
+    if (!precise) {
+        if (need_scan) {
+            fprintf(stderr,
```
Comment (neverchanje) [inline src/shell/commands/data_operations.cpp:2217]: So if the user specifies `--partition`, she must specify `--precise` too?

File: src/shell/commands/data_operations.cpp (hunk 5 above)
Comment (Shuo-Jia) [reply src/shell/commands/data_operations.cpp:2131]: Inputting any valid parameter will means want precise result, not only  `[-p|--partition] or [-c|--precise]`

File: src/shell/commands/data_operations.cpp (hunk 6 above)
Comment (Shuo-Jia) [reply src/shell/commands/data_operations.cpp:2217]: Yeah, just for simplicity. For `estimate count` can be get  immediately, it's not necessary to filter `partition`

File: src/shell/commands/data_operations.cpp [hunk 7]
```diff
@@ -2201,6 +2212,42 @@ bool count_data(command_executor *e, shell_context *sc, arguments args)
...
+                    "ERROR: you must input [-c|-pricise] flag when you expect to get pricise "
```
Comment (hycdong) [inline src/shell/commands/data_operations.cpp:2218]: [-c|--precise]

**Pull request merged**